broker/src/test/java/io/camunda/zeebe/broker/system/partitions/AsyncSnapshotingOnFollowerTest.java

```diff
@@ -0,0 +1,120 @@
+
+  private final ControlledActorClock clock = new ControlledActorClock();
+  private final ActorSchedulerRule actorSchedulerRule = new ActorSchedulerRule(clock);
+
+  @Rule
+  public final RuleChain chain =
+      RuleChain.outerRule(autoCloseableRule).around(tempFolderRule).around(actorSchedulerRule);
+
+  private AsyncSnapshotDirector asyncSnapshotDirector;
+  private StreamProcessor mockStreamProcessor;
+  private ConstructableSnapshotStore persistedSnapshotStore;
+  private StateControllerImpl snapshotController;
+
+  @Before
+  public void setup() throws IOException {
+    final var rootDirectory = tempFolderRule.getRoot().toPath();
+    final var factory = new FileBasedSnapshotStoreFactory(actorSchedulerRule.get(), 1);
+    final int partitionId = 1;
+
+    factory.createReceivableSnapshotStore(rootDirectory, partitionId);
+    persistedSnapshotStore = factory.getConstructableSnapshotStore(partitionId);
+
+    createSnapshotController(rootDirectory, factory, partitionId);
+
+    final ActorScheduler actorScheduler = actorSchedulerRule.get();
+    createStreamProcessorControllerMock();
+    createAsyncSnapshotDirector(actorScheduler);
+  }
+
+  private void createSnapshotController(
+      final Path rootDirectory,
+      final FileBasedSnapshotStoreFactory factory,
+      final int partitionId) {
+    snapshotController =
+        new StateControllerImpl(
+            1,
+            ZeebeRocksDbFactory.newFactory(),
+            persistedSnapshotStore,
+            factory.getReceivableSnapshotStore(partitionId),
+            rootDirectory.resolve("runtime"),
+            new NoneSnapshotReplication(),
+            l ->
+                Optional.of(
+                    new TestIndexedRaftLogEntry(
+                        l + 100, 1, new ApplicationEntry(1, 10, new UnsafeBuffer()))),
+            db -> Long.MAX_VALUE);
+
+    snapshotController.openDb();
+    autoCloseableRule.manage(snapshotController);
+  }
+
+  private void createStreamProcessorControllerMock() {
+    mockStreamProcessor = mock(StreamProcessor.class);
+
+    when(mockStreamProcessor.getLastProcessedPositionAsync())
+        .thenReturn(CompletableActorFuture.completed(10L));
+
+    when(mockStreamProcessor.getLastWrittenPositionAsync())
+        .thenReturn(CompletableActorFuture.completed(15L));
+  }
+
+  private void createAsyncSnapshotDirector(final ActorScheduler actorScheduler) {
+    asyncSnapshotDirector =
+        AsyncSnapshotDirector.ofReplayMode(
+            0, 1, mockStreamProcessor, snapshotController, Duration.ofMinutes(1));
+    actorScheduler.submitActor(asyncSnapshotDirector).join();
+  }
+
+  @Test
+  public void shouldPersistSnapshotWithoutWaitingForCommit() {
+    // when
+    asyncSnapshotDirector.forceSnapshot();
+
+    // then
+    waitUntil(() -> snapshotController.getValidSnapshotsCount() == 1);
+    assertThat(snapshotController.getValidSnapshotsCount()).isEqualTo(1);
+  }
```

Comment:
Can we merge the test case to `AsyncSnapshotingTest`?

The setup code looks very similar to the existing test. I'm not sure if it is worth copying so much code for one test case.